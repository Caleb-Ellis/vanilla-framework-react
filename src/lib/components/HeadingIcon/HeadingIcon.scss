@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_heading-icon';
@include vf-base;
@include vf-p-heading-icon;
