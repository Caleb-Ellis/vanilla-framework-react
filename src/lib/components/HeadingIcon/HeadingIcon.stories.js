import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import HeadingIcon from './HeadingIcon';

storiesOf('Heading Icon', module)
  .add('Default',
    withInfo('The HeadingIcon component can be used to add an icon to a standard header.')(() => (
      <HeadingIcon
        title="Lorem ipsum dolor"
        alt="Lorem ipsum dolor"
        src="https://assets.ubuntu.com/v1/a4d31b28-icon-quote.svg"
      />),
    ),
  )

  .add('With Children',
    withInfo('The HeadingIcon component can be used to add an icon to a standard header.')(() => (
      <HeadingIcon
        title="Lorem ipsum dolor"
        alt="Lorem ipsum dolor"
        src="https://assets.ubuntu.com/v1/a4d31b28-icon-quote.svg"
      >
        <p>
          sit amet, consectetur adipisicing elit. Enim excepturi, repudiandae blanditiis odio
          perferendis voluptatibus dolorum, dicta illum quae ipsa voluptatum, sunt quasi? Nulla
          reiciendis magnam nostrum aliquam, beatae doloribus.
        </p>
      </HeadingIcon>),
    ),
  );
