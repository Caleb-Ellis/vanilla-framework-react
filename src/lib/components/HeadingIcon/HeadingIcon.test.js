import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import HeadingIcon from './HeadingIcon';

describe('HeadingIcon component', () => {
  it('should render an icon and title without children correctly', () => {
    const headingIcon = ReactTestRenderer.create(
      <HeadingIcon
        title="Lorem ipsum dolor"
        alt="Lorem ipsum dolor"
        src="https://assets.ubuntu.com/v1/a4d31b28-icon-quote.svg"
      />,
    );
    const json = headingIcon.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    const headingIcon = ReactTestRenderer.create(
      <HeadingIcon
        title="Lorem ipsum dolor"
        alt="Lorem ipsum dolor"
        src="https://assets.ubuntu.com/v1/a4d31b28-icon-quote.svg"
      >
        <p>
          sit amet, consectetur adipisicing elit. Enim excepturi, repudiandae blanditiis odio
          perferendis voluptatibus dolorum, dicta illum quae ipsa voluptatum, sunt quasi? Nulla
          reiciendis magnam nostrum aliquam, beatae doloribus.
        </p>
      </HeadingIcon>,
    );
    const json = headingIcon.toJSON();
    expect(json).toMatchSnapshot();
  });
});
