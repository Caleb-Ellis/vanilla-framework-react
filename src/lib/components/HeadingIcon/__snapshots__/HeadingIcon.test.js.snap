// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingIcon component should render an icon and title without children correctly 1`] = `
<div
  className="p-heading-icon"
>
  <div
    className="p-heading-icon__header"
  >
    <img
      alt="Lorem ipsum dolor"
      className="p-heading-icon__img"
      src="https://assets.ubuntu.com/v1/a4d31b28-icon-quote.svg"
    />
    <h3
      className="p-heading-icon__title"
    >
      Lorem ipsum dolor
    </h3>
  </div>
  
</div>
`;

exports[`HeadingIcon component should render children correctly 1`] = `
<div
  className="p-heading-icon"
>
  <div
    className="p-heading-icon__header"
  >
    <img
      alt="Lorem ipsum dolor"
      className="p-heading-icon__img"
      src="https://assets.ubuntu.com/v1/a4d31b28-icon-quote.svg"
    />
    <h3
      className="p-heading-icon__title"
    >
      Lorem ipsum dolor
    </h3>
  </div>
  <p>
    sit amet, consectetur adipisicing elit. Enim excepturi, repudiandae blanditiis odio perferendis voluptatibus dolorum, dicta illum quae ipsa voluptatum, sunt quasi? Nulla reiciendis magnam nostrum aliquam, beatae doloribus.
  </p>
</div>
`;
